src/libgeds/TcpTransport.cpp

```diff
@@ -65,7 +65,7 @@ void TcpTransport::start() {
   num_proc = std::min(num_proc, MAX_IO_THREADS);
   isServing = true;
 
-  for (unsigned int id = 0; id < MAX_IO_THREADS; id++) {
+  for (unsigned int id = 0; id < num_proc; id++) {
```

Comment:
I thought we may even want more io threads than processor cores? especially useful on systems with a low core count. think of a single core system where we may want to avoid head of line blocking of a single tcp connection